docs: do no mention cross compile (#51202)
LLNL/spack: lib/spack/docs/features.rst

--------------------------------

Spack allows installation to be customized.

> Users can specify the version, compile-time options, and target architecture for each package.

Author: ```suggestion
Users can specify the version, compile-time options, and target architecture, all on the command line.
```
